src/frontend/packages/core/src/shared/components/list/list-types/endpoint/endpoint-list.helpers.ts

```diff
@@ -132,11 +135,23 @@ export class EndpointListHelper {
         },
         label: 'Connect',
         description: '',
-        createVisible: (row$: Observable<EndpointModel>) => row$.pipe(map(row => {
-          const endpoint = entityCatalog.getEndpoint(row.cnsi_type, row.sub_type);
-          const ep = endpoint ? endpoint.definition : { unConnectable: false };
-          return !ep.unConnectable && row.connectionStatus === 'disconnected';
-        }))
+        createVisible: (row$: Observable<EndpointModel>) => {
+          return combineLatest([
+            this.sessionService.userEndpointsNotDisabled(),
+            this.currentUserPermissionsService.can(StratosCurrentUserPermissions.EDIT_ADMIN_ENDPOINT),
+            row$
+          ]).pipe(
+            map(([userEndpointsEnabled, isAdmin, row]) => {
+              if (userEndpointsEnabled && !row.creator.admin && isAdmin){
```

Comment:
This is hiding the connect button in user-endpoints for admins, so that admins can't connect to user created endpoints. Since users can't see endpoints created by other users, they are allowed to create an endpoint with the same APIUrl from another endpoint. Admins on the other side can see all user-endpoints. If I'm allowing admins to connect to user-endpoints, that would open up the possibility for admins to connect to the same APIUrl two or more times through different user-endpoints.